build(docs-infra): update Angular framework/Material to versions 13.0.0-next.9/7

gkalpak opened the pull request:
See individual commits for details.

<sub>This is the equivalent of #43687, but for the master branch.</sub>

##
Fixes #28114.

mary-poppins commented:
You can preview 8c90794 at https://pr43686-8c90794.ngbuilds.io/.

mary-poppins commented:
You can preview 4b22c3b at https://pr43686-4b22c3b.ngbuilds.io/.

mary-poppins commented:
You can preview 2a9f6a7 at https://pr43686-2a9f6a7.ngbuilds.io/.

mary-poppins commented:
You can preview fcfb4d1 at https://pr43686-fcfb4d1.ngbuilds.io/.

crisbeto commented:
FYI I did this already in https://github.com/angular/angular/pull/43642.

mary-poppins commented:
You can preview 3f7abe9 at https://pr43686-3f7abe9.ngbuilds.io/.

gkalpak commented:
Thx for the heads-up, @crisbeto. I'll rebase this PR if yours gets merged first.
(For now, I am keeping the changes here as well in case this gets merged before your PR, because we want to get the updates deployed to angular.io asap.)

mary-poppins commented:
You can preview 83722ff at https://pr43686-83722ff.ngbuilds.io/.

petebacondarwin reviewed (approved):
Reviewed-for: size-tracking

mary-poppins commented:
You can preview a8022de at https://pr43686-a8022de.ngbuilds.io/.

AndrewKushnir commented on goldens/size-tracking/aio-payloads.json:
```diff
@@ -3,23 +3,23 @@
       "uncompressed": {
         "runtime": 4454,
-        "main": 460460,
+        "main": 462288,
```
@gkalpak just curious if we know where this extra ~1.8kb is coming from?

gkalpak replied:
<sub>You mean 1.17kb, right? 😛</sub>

Not sure. It is not unusual for Angular Material/CDK component sizes to increase a bit in each release (I assume due to new fixes/features). Since it is a small change (~0.25% of the main bundle size), I didn't look into what exactly is causing it.

AndrewKushnir reviewed (approved):
Reviewed-for: size-tracking

mary-poppins commented:
You can preview 31471eb at https://pr43686-31471eb.ngbuilds.io/.

gkalpak commented:
The CI failures will be fixed by #43714.

mary-poppins commented:
You can preview baaf1c6 at https://pr43686-baaf1c6.ngbuilds.io/.

gkalpak committed:
build(docs-infra): use `$schema` in `ngsw-config.template.json` to provide intellisense

Add an appropriate `$schema` property in `ngsw-config.template.json` to
allow IDEs to provide intellisense about the various fields of the
config.

gkalpak committed:
refactor(docs-infra): prefix error messages with Angular version info

Prefix error messages with the Angular version currently used in the
angular.io app. This applies to both errors logged to the console and
reported to Google Analytics and makes debugging certain errors easier.

gkalpak committed:
build(docs-infra): update AIO to align with new v13.0.0-next.9 CLI apps

This commit updates angular.io to more closely align with new apps
created with the latest Angular CLI and remove redundant files/config
now that CLI has dropped support for differential loading.

gkalpak committed:
build(docs-infra): update Angular Material to version 13.0.0-next.7

This commit updates angular.io to the latest prerelease version of
Angular CDK/Material (v13.0.0-next.7).

gkalpak committed:
build(docs-infra): ensure `ngsw.json` generation is correct and deterministic

As explained in #43679, currently the generation of the `ngsw.json` SW
manifest is non-deterministic and often incorrect. Until we can update
to an `@angular/service-worker` version that includes the fix from
PR #43679, we temporarily work-around the issue by re-generating the
`ngsw.json` manifest after `ng build` using the `ngsw-config` binary
exposed by `@angular/service-worker`.

NOTE:
This works around the issue, because the [FileSystem][1] class used by
the `ngsw-config` binary happens to be synchronous (unlike the
implementation provided by the Angular CLI), thus avoiding the race
conditions described in #43679.

[1]: https://github.com/angular/angular/blob/c721135e370b34c840756bcfb22c8119b4c8c452/packages/service-worker/cli/filesystem.ts#L15

gkalpak committed:
fixup! refactor(docs-infra): prefix error messages with Angular version info

dylhunn commented:
This PR was merged into the repository by commit 841c5aebbfc7f40fdf838cc3e73395cf8b94bdf9.
